R/Viewer.R

```diff
@@ -98,8 +98,8 @@ odbcListObjects.OdbcConnection <- function(connection, catalog = NULL, schema =
   # if no schema was supplied but this database has schema, return a list of
   # schema
   if (is.null(schema)) {
-    schemas <- string_values(connection_sql_tables(connection@ptr, "", schema_name = schema %||% "%", "", NULL)[["table_schema"]])
```

Comment:
Returning only non-zero values is what `string_values` was written for, so you don't need an extra test.